src/hotspot/share/gc/shared/cardTable.cpp

```diff
@@ -262,3 +274,220 @@ void CardTable::print_on(outputStream* st) const {
   st->print_cr("Card table byte_map: [" PTR_FORMAT "," PTR_FORMAT "] _byte_map_base: " PTR_FORMAT,
                p2i(_byte_map), p2i(_byte_map + _byte_map_size), p2i(_byte_map_base));
 }
+
+MemRegion CardTable::committed_for_region_in_shared_virtual_space(const MemRegion mr) const {
+  HeapWord* addr_l = (HeapWord*)byte_for(mr.start());
+
+  log_trace(gc, barrier)("CardTable::committed_for_region_in_shared_virtual_space: ");
+  log_trace(gc, barrier)("    mr.start():                          " PTR_FORMAT "  mr.last():                          " PTR_FORMAT,
+                         p2i(mr.start()), p2i(mr.last()));
+  log_trace(gc, barrier)("    byte_for(mr.start()):                " PTR_FORMAT, p2i(addr_l));
+
+  if (mr.start() == _covered[0].start()) {
+    addr_l = (HeapWord*)align_down(addr_l, _page_size);
+    log_trace(gc, barrier)("    aligned byte_for(mr.start()):        " PTR_FORMAT, p2i(addr_l));
+  }
+
+  HeapWord* addr_r;
+  if (mr.is_empty()) {
+    addr_r = addr_l;
+  } else {
+    addr_r = (HeapWord*)byte_after(mr.last());
+    log_trace(gc, barrier)("    byte_after(mr.last()):               " PTR_FORMAT, p2i(addr_r));
+
+    if (mr.start() == _covered[1].start()) {
+      addr_r = (HeapWord*)align_up(addr_r, _page_size);
+      log_trace(gc, barrier)("    aligned byte_after(mr.last()):       " PTR_FORMAT, p2i(addr_r));
+    }
+  }
+
+  return MemRegion(addr_l, addr_r);
+}
+
+void CardTable::resize_covered_region_in_shared_virtual_space(MemRegion new_region0, MemRegion new_region1) {
```

Comment:
Maybe it's necessary to stay in line with historical code, but it would be better if variable names were clearer about whether they were referring to card table memory or heap memory.

Reply:
> Maybe it's necessary to stay in line with historical code, but it would be better if variable names were clearer about whether they were referring to card table memory or heap memory.

This is a good suggestion, I have renamed the 2 parameters. I'm reviewing the other variables to see where additional renaming will be helpful